CHANGELOG.md

```diff
@@ -52,6 +52,7 @@ The next release will require at least [Go 1.23].
 
 ### Fixes
 
+- Upgrade cenkalti/backoff from v4 to v5, fix the breaking changes in that version, and upgrade to it (#6417)
```

Comment:
Unless this upgrade causes user-visible changes (which it ideally shouldn't), this doesn't need a changelog entry.